include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -131,10 +93,72 @@ namespace SuperiorMySqlpp
              typename ConnType>
     void psReadQuery(const std::string &query, ConnType &&connection, Callable &&processingFunction)
     {
-        auto ps = detail::generatePreparedStatementImpl<storeResult, validateMode, warnMode, ignoreNullable>(connection, query, processingFunction);
+        auto ps = detail::generatePreparedStatement<storeResult, validateMode, warnMode, ignoreNullable>(connection, query, processingFunction, [](){});
         psReadQuery(ps, processingFunction);
     }
 
+    /**
+     * @brief Builds prepared statement from query string and constructs prepared stament with updatable arguments
+     * @param query Query to be executed (in most cases there will be selections)
+     * @param connection Connection handle into database
+     * @param paramsSetter function setting statement's parameters, returning bool if input data are available
+     * @param processingFunction function to be invoked on every row
+     *                           Its parameters must correspond with result columns (their types and count)
+     * @tparam storeResult Boolean indicating if results will be in `store` or `use` mode
+     * @tparam validateMode Indicates validate mode level
+     * @tparam warnMode Indicates warning mode level
+     * @tparam ignoreNullable Disables null type checking
+     */
+    template<bool storeResult=detail::PreparedStatementsDefault::getStoreResult(),
+             ValidateMetadataMode validateMode=detail::PreparedStatementsDefault::getValidateMode(),
+             ValidateMetadataMode warnMode=detail::PreparedStatementsDefault::getWarnMode(),
+             bool ignoreNullable=detail::PreparedStatementsDefault::getIgnoreNullable(),
+             typename ResultCallable,
+             typename ParamCallable,
+             typename ConnType>
+    void psQuery(const std::string &query, ConnType &&connection, ParamCallable &&paramsSetter, ResultCallable &&processingFunction)
```

Comment:
I'd attempt to unify implementations of ps(Read)Query variants if at all possible - the shared implementation would take paramUpdater and resultReader callbacks along with `bool`s that could be used to conditionally execute the callbacks.
Only reason why that wouldn't be a good idea is if compiler wouldn't be able to optimize that away.